server/src/test/java/io/crate/integrationtests/DiskThresholdDeciderIT.java

```diff
@@ -0,0 +1,365 @@
+                .put(Environment.PATH_DATA_SETTING.getKey(), dataPath)
+                .put(FsService.ALWAYS_REFRESH_SETTING.getKey(), true)
+                .put(DiskThresholdSettings.CLUSTER_ROUTING_ALLOCATION_LOW_DISK_WATERMARK_SETTING.getKey(), WATERMARK_BYTES + "b")
+                .put(DiskThresholdSettings.CLUSTER_ROUTING_ALLOCATION_HIGH_DISK_WATERMARK_SETTING.getKey(), WATERMARK_BYTES + "b")
+                .put(DiskThresholdSettings.CLUSTER_ROUTING_ALLOCATION_DISK_FLOOD_STAGE_WATERMARK_SETTING.getKey(), "0b")
+                .put(DiskThresholdSettings.CLUSTER_ROUTING_ALLOCATION_REROUTE_INTERVAL_SETTING.getKey(), "0ms")
+                .build();
+    }
+
+    @Override
+    protected Collection<Class<? extends Plugin>> nodePlugins() {
+        return Collections.singletonList(InternalSettingsPlugin.class);
+    }
+
+    public void testHighWatermarkNotExceeded() throws InterruptedException, ExecutionException {
+        internalCluster().startMasterOnlyNode();
+        internalCluster().startDataOnlyNode();
+        final String dataNodeName = internalCluster().startDataOnlyNode();
+
+        final InternalClusterInfoService clusterInfoService
+                = (InternalClusterInfoService) internalCluster().getMasterNodeInstance(ClusterInfoService.class);
+        internalCluster().getMasterNodeInstance(ClusterService.class).addListener(event -> clusterInfoService.refresh());
+
+        final String dataNode0Id = internalCluster().getInstance(NodeEnvironment.class, dataNodeName).nodeId();
+        final Path dataNode0Path = internalCluster().getInstance(Environment.class, dataNodeName).dataFiles()[0];
+
+        String indexName = "test";
+        createIndex(indexName); // creates a table doc.test without columns.
+
+        final long minShardSize = createReasonableSizedShards(indexName);
+
+        // reduce disk size of node 0 so that no shards fit below the high watermark, forcing all shards onto the other data node
+        // (subtract the translog size since the disk threshold decider ignores this and may therefore move the shard back again)
+        fileSystemProvider.getTestFileStore(dataNode0Path).setTotalSpace(minShardSize + WATERMARK_BYTES - 1L);
+        refreshDiskUsage();
+        assertThat(getShardRoutings(dataNode0Id, indexName)).isEmpty();
+
+        // increase disk size of node 0 to allow just enough room for one shard, and check that it's rebalanced back
+        fileSystemProvider.getTestFileStore(dataNode0Path).setTotalSpace(minShardSize + WATERMARK_BYTES + 1L);
+        refreshDiskUsage();
+        assertThat(getShardRoutings(dataNode0Id, indexName)).hasSize(1);
+    }
+
+    private Set<ShardRouting> getShardRoutings(String nodeId, String indexName) {
+        final Set<ShardRouting> shardRoutings = new HashSet<>();
+        ClusterStateResponse clusterStateResponse = FutureUtils.get(client().admin().cluster().state(new ClusterStateRequest().routingTable(true)));
```

Comment:
We removed `prepareState` in https://github.com/crate/crate/commit/f0f59ea4dbbb14677cb6e1865617f403984b652d so using request directly